hack/lib/kube-versions.txt

```diff
@@ -1,9 +1,12 @@
 # This file lists the versions of Kubernetes to use for codegen.
 # They must be listed newest to oldest.
-1.29.2
-1.28.7
-1.27.11
-1.26.14
+# New releases are listed at https://kubernetes.io/releases/.
+# Older releases are listed at https://kubernetes.io/releases/patch-releases/.
+1.30.0
+1.29.4
+1.28.9
+1.27.13
+1.26.15
 1.25.16
```

Comment:
Do we want to get rid of 1.25.x codegen?

Reply:
I think we had discussed removing 1.21 and 1.22 support since those versions don't recognize `x-kubernetes-validations` on CRDs. I think we're ok to keep 1.23/1.24/1.25.

Reply:
I don't think it matters if the old versions don't recognize `x-kubernetes-validations`, does it? We don't use the generated code inside our own project for except for the most recent version.

I think we can decide to remove old version of codegen based only on how long that version of Kube has been out of support. Support for 1.22 ended 1.5 years ago (28 Oct 2022), so agreed that it seems like a conservative cutoff and we should remove 1.22 and older.

Support for 1.23 ended just over a year ago. Maybe we can also remove that version?

All other versions are either still in support, or support ended less than a year ago. Maybe that's a good rule of thumb for which versions to keep?

Reply:
Removing some versions could be a separate PR, due to the chicken and egg problem between the branch and what versions the verify-codegen CI job wants.

Reply:
I updated the definition of the verify-codegen CI job to make it more lenient to the case where a version has been removed from `kube-versions.txt`. We can try removing the old versions in this PR and see if my change works.

Reply:
I removed 1.21, 1.22, and 1.23 in this PR.